Kudu.Services/ServiceHookHandlers/GitlabHqHandler.cs

```diff
@@ -7,68 +7,81 @@ namespace Kudu.Services.ServiceHookHandlers
 {
     public class GitlabHqHandler : GitHubCompatHandler
     {
+        //{
+        //  "before": "2370e44c850e732d71edd2db36920482558e3fe0",
+        //  "after": "2370e44c850e732d71edd2db36920482558e3fe0",
+        //  "ref": "refs/heads/master",
+        //  "checkout_sha": "2370e44c850e732d71edd2db36920482558e3fe0",
+        //  "user_id": 99630,
+        //  "user_name": "Suwat Bodin",
+        //  "project_id": 171368,
+        //  "repository": {
+        //    "name": "HelloKudu",
+        //    "url": "git@gitlab.com:KuduApps/HelloKudu.git",
+        //    "description": "",
+        //    "homepage": "https://gitlab.com/KuduApps/HelloKudu",
+        //    "git_http_url": "https://gitlab.com/KuduApps/HelloKudu.git",
+        //    "git_ssh_url": "git@gitlab.com:KuduApps/HelloKudu.git",
+        //    "visibility_level": 20
+        //  },
+        //  "commits": [
+        //    {
+        //      "id": "2370e44c850e732d71edd2db36920482558e3fe0",
+        //      "message": "Initial commit\n",
+        //      "timestamp": "2012-02-14T10:45:08-08:00",
+        //      "url": "https://gitlab.com/KuduApps/HelloKudu/commit/2370e44c850e732d71edd2db36920482558e3fe0",
+        //      "author": {
+        //        "name": "David Fowler",
+        //        "email": "davidfowl@gmail.com"
+        //      }
+        //    }
+        //  ],
+        //  "total_commits_count": 1
+        //}
+
+        private const int PublicVisibilityLevel = 20;
+
         protected override GitDeploymentInfo GetDeploymentInfo(HttpRequestBase request, JObject payload, string targetBranch)
         {
             var repository = payload.Value<JObject>("repository");
             var userid = payload.Value<int?>("user_id");
             var username = payload.Value<string>("user_name");
-
             if (repository == null || userid == null || username == null)
             {
                 // doesn't look like GitlabHQ
                 return null;
             }
 
-            string newRef = payload.Value<string>("after");
-            if (IsDeleteCommit(newRef))
+            var url = repository.Value<string>("url");
+            if (url == null || url.IndexOf("gitlab.com", StringComparison.OrdinalIgnoreCase) < 0)
```

Comment:
I think we want to avoid testing the domain name, as this will block and other gitlab deployment (i.e. like the GitHub enterprise scenario). So it's best to sniff it in some other way.